### PR TITLE
[Paddle TensorRT] Add converter layer naming function set_layer_name

### PR DESCRIPTION
**Pull request by PolaKuma**
### PR Category
User Experience

### PR Types
New features

### Description
card-71500
- 新增给converter输出层命名函数set_layer_name，格式：id_pd_op.xxx->layerName(inputId)
- 两种用法：
1. 对于不是公共函数，传入 (layer: ILayer, paddle_op: op) 
2. 对于公共函数，函数新增name=None参数，并传入(layer: ILayer, [paddle_op.name(), layerName]: List)
3. 对于公共函数嵌套，同（二），但在外层嵌套的公共函数加一个 if name != None 的判断

eg. 对于gelu:
Screenshot 2025-02-20 at 10 50 22

常用算子命名：add、add_、relu、matmul、full_int_array、reshape、transpose、full、scale、softmax、layer_norm、conv2d、nonzero、gather_nd、pool2d、batch_norm、batch_norm_、flatten、concat、gelu、unsqueeze、unsqueeze_、squeeze、squeeze_

### CODE REVIEW

File: python/paddle/tensorrt/converter_utils.py [hunk 1]
```diff
@@ -793,3 +798,53 @@ def WithFp16():
...
+        else:
+            # Handling for layer
+            op_name = second_param.name()
+            layer_var_name = None
+            if op_name is not None:
+                # Retrieve the name of the variable that refers to the layer
+                for (
+                    var_name,
+                    var_val,
+                ) in inspect.currentframe().f_back.f_locals.items():
```
Comment (lizexu123) [inline python/paddle/tensorrt/converter_utils.py:927]: 我的建议是不用inspect，直接用layer.type.name来代替，因为本身前面就有pd_op.gelu(序号)，就可以分辨哪一个layer了

Comment (YuanRisheng) [reply python/paddle/tensorrt/converter_utils.py:927]: 这个我建议还是用局部变量，这个能更快区分出是哪个layer

Comment (PolaKuma) [reply python/paddle/tensorrt/converter_utils.py:927]: done

**Review by lizexu123 — DISMISSED** [review]

File: python/paddle/tensorrt/converter_utils.py [hunk 2]
```diff
@@ -789,3 +822,53 @@ def WithFp16():
         enable_fp16 = True
     # TODO(lizexu123) WithInt8() and use_dla are not yet implemented
     return enable_fp16
+
+
+def set_layer_name(layer, second_param):
```
Comment (YuanRisheng) [inline python/paddle/tensorrt/converter_utils.py:895]: 给这个函数再加点描述吧，说明一下命名规则

Comment (PolaKuma) [reply python/paddle/tensorrt/converter_utils.py:895]: done

**Review by lizexu123 — APPROVED** [review]
